daemon/containerd/progress.go

```diff
@@ -51,6 +51,10 @@ func (j *jobs) showProgress(ctx context.Context, out progress.Output, updater pr
 					return
 				}
 			case <-ctx.Done():
+				ctx, err := context.WithTimeout(ctx, time.Millisecond*100)
```

Comment:
if `ctx` is done, shouldn't the new conext be done as well?

Reply:
`UpdateProgress` isn't doing anything with the context (though it should).
There is a call to `ListStatuses` on the containerd client so I'm suprised that even works since that needs to go the grpc API.

I also think 100ms may be a bit short, particularly because there is an RPC involved.

Reply:
Yes! That was the intention here, but I somehow missed it that I reused the same context 🤦🏻 
Thanks for noticing. Not sure why it worked too, I thought that maybe `WithTimeout` refreshes the context and makes it un-done, but I just tested it and that's not it: https://go.dev/play/p/Znmqr0V32OJ?v=goprev.
Looks like I was just (un)lucky enough to not notice the same incomplete progress with this change.

Reply:
Increased the timeout to 500ms.

Reply:
Also, I just realized that this is not enough - we call UpdateProgress with fresh context, but the `progress.Output` stream may already by closed by the caller because we handle this in a separate goroutine.
I made the cancel function returned from `showProgress` wait for the last update to finish. It's a bit ugly, but it works correctly now.
Let me know if you see a better way to handle this.